pkg/cloud/ibmcloud/clients/machines.go

```diff
@@ -74,28 +78,33 @@ func NewInstanceServiceFromMachine(kubeClient kubernetes.Interface, machine *clu
 	return NewGuestService(sess), nil
 }
 
-func (gs *GuestService) waitGuestReady(Id int) {
+func (gs *GuestService) waitGuestReady(Id int) error {
 	// Wait for transactions to finish
 	klog.Info("Waiting for transactions to complete before destroying.")
 	s := services.GetVirtualGuestService(gs.sess).Id(Id)
 
 	// Delay to allow transactions to be registered
-	time.Sleep(5 * time.Second)
+	time.Sleep(RetryIntervalWaitReady)
 
 	// TODO: make it V(2) after initial development work done if necessary
 	//if klog.V(2) {
 	// Enable debug to show messages from IBM Cloud during node provision
 	s.Session.Debug = true
 	//}
+	sum := RetryIntervalWaitReady
 	for transactions, _ := s.GetActiveTransactions(); len(transactions) > 0; {
-		// TODO(gyliu513) make it configurable or use the notification mechanism to optimize
-		// the process instead of hardcoded waiting.
-		time.Sleep(5 * time.Second)
+		time.Sleep(RetryIntervalWaitReady)
+		sum += RetryIntervalWaitReady
+		if sum > TimeoutWaitReady {
+			// Now the guest failed to reach timeout
+			return fmt.Errorf("Timeout wait for guest %d ready", Id)
```

Comment:
Let's make message consistent. 
s/Timeout wait for guest %d ready/Waiting for guest %d ready time out/